src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -851,7 +902,7 @@ PtrCouplingScheme CouplingSchemeConfiguration::createSerialImplicitCouplingSchem
       accessor, m2n, _config.dtMethod, BaseCouplingScheme::Implicit, _config.maxIterations, _config.extrapolationOrder);
 
   addDataToBeExchanged(*scheme, accessor);
-  PRECICE_CHECK(scheme->hasAnySendData(),
+  PRECICE_CHECK(scheme->hasAnySendData() || scheme->hasAnySendGlobalData(),
```

Comment:
Are really both these functions required? Or could we merge them?

Reply:
I kept them separated to maintain some kind of modularity. But yes can be merged.